src/sqlfluff/core/dialects/dialect_snowflake.py

```diff
@@ -561,7 +563,7 @@ class ShowStatementSegment(BaseSegment):
         OneOf("TERSE", optional=True),
         _object_types_plural,
         OneOf("HISTORY", optional=True),
-        Sequence("LIKE", Ref("ObjectReferenceSegment"), optional=True),
+        Sequence("LIKE", Ref("Expression_A_Grammar"), optional=True),
```

Comment:
How come Expression_A_Grammar not ObjectReferenceSegment?

Reply:
It is really a string expression. Possible values could be `LIKE '%MYTABLE%'`. I guess we could use `QuotedLiteralSegment`, which does pass the tests. What do you think?

Reply:
I think that's a better suggestion. Expression_A_Grammar is a very flexible grammar which doesn't need to be used here so I'd keep it simple with `QuotedLiteralSegment`.